pkg/cmd/pulumi/import.go

```diff
@@ -212,6 +212,42 @@ func parseImportFile(f importFile, protectResources bool) ([]deploy.Import, impo
 		errs = multierror.Append(errs, fmt.Errorf(format, args...))
 	}
 
+	makeUnique, checkAmbiguous := func() (func(int, tokens.QName) tokens.QName, func(tokens.QName) bool) {
+		// Track used resource names.
+		takenNames := map[tokens.QName]struct{}{}
+		// Track indexes that are not unique and need to be made unique.
+		duplicateIndexes := map[int]struct{}{}
+		// Track parent/provider/etc references that are ambiguous when referenced.
+		ambiguousNames := map[tokens.QName]struct{}{}
+		for i, spec := range f.Resources {
+			if _, exists := takenNames[spec.Name]; exists {
+				duplicateIndexes[i] = struct{}{}
+				ambiguousNames[spec.Name] = struct{}{}
+			}
+			// Prepopulate already taken names first to avoid using them in makeUnique.
+			takenNames[spec.Name] = struct{}{}
+		}
+		checkAmbiguous := func(name tokens.QName) bool {
+			_, isAmbiguous := ambiguousNames[name]
+			return isAmbiguous
+		}
+		makeUnique := func(i int, name tokens.QName) tokens.QName {
+			if _, isDuplicate := duplicateIndexes[i]; isDuplicate {
```

Comment:
This seems wrong? If it is a duplicate we just return it?

Reply:
added the other tests you suggested with explicit names which caught this. nice catch. forgot the `!`.